x-pack/plugins/index_lifecycle_management/public/application/sections/edit_policy/form/serializer.ts

```diff
@@ -63,123 +75,137 @@ const serializeAllocateAction = (
 export const createSerializer = (originalPolicy?: SerializedPolicy) => (
   data: FormInternal
 ): SerializedPolicy => {
-  const { _meta, ...policy } = data;
+  const { _meta, ...updatedPolicy } = data;
 
-  if (!policy.phases || !policy.phases.hot) {
-    policy.phases = { hot: { actions: {} } };
+  if (!updatedPolicy.phases || !updatedPolicy.phases.hot) {
+    updatedPolicy.phases = { hot: { actions: {} } };
   }
 
-  /**
-   * HOT PHASE SERIALIZATION
-   */
-  if (policy.phases.hot) {
-    policy.phases.hot.min_age = originalPolicy?.phases.hot?.min_age ?? '0ms';
-  }
+  return produce<SerializedPolicy>(originalPolicy ?? defaultPolicy, (draft) => {
+    // Copy over all updated fields
+    merge(draft, updatedPolicy);
 
-  if (policy.phases.hot?.actions) {
-    if (policy.phases.hot.actions?.rollover && _meta.hot.useRollover) {
-      if (policy.phases.hot.actions.rollover.max_age) {
-        policy.phases.hot.actions.rollover.max_age = `${policy.phases.hot.actions.rollover.max_age}${_meta.hot.maxAgeUnit}`;
-      }
+    // Next copy over all meta fields and delete any fields that have been removed
+    // by fields exposed in the form. It is very important that we do not delete
+    // data that the form does not control! E.g., unfollow action in hot phase.
+
+    /**
+     * HOT PHASE SERIALIZATION
+     */
+    if (draft.phases.hot) {
+      draft.phases.hot.min_age = draft.phases.hot.min_age ?? '0ms';
+    }
+
+    if (draft.phases.hot?.actions) {
+      const hotPhaseActions = draft.phases.hot.actions;
+      if (hotPhaseActions.rollover && _meta.hot.useRollover) {
+        if (hotPhaseActions.rollover.max_age) {
+          hotPhaseActions.rollover.max_age = `${hotPhaseActions.rollover.max_age}${_meta.hot.maxAgeUnit}`;
+        }
+
+        if (hotPhaseActions.rollover.max_size) {
+          hotPhaseActions.rollover.max_size = `${hotPhaseActions.rollover.max_size}${_meta.hot.maxStorageSizeUnit}`;
+        }
 
-      if (policy.phases.hot.actions.rollover.max_size) {
-        policy.phases.hot.actions.rollover.max_size = `${policy.phases.hot.actions.rollover.max_size}${_meta.hot.maxStorageSizeUnit}`;
+        if (_meta.hot.bestCompression && hotPhaseActions.forcemerge) {
+          hotPhaseActions.forcemerge.index_codec = 'best_compression';
+        }
+      } else {
+        delete hotPhaseActions.rollover;
+        delete hotPhaseActions.forcemerge;
       }
 
-      if (_meta.hot.bestCompression && policy.phases.hot.actions?.forcemerge) {
-        policy.phases.hot.actions.forcemerge.index_codec = 'best_compression';
+      if (!updatedPolicy.phases.hot!.actions?.set_priority && hotPhaseActions.set_priority) {
```

Comment:
Seems like this could be simplified to:

```js
if (!updatedPolicy.phases.hot!.actions?.set_priority) {
  delete hotPhaseActions.set_priority;
}
```

It might be personal preference, but the second condition makes me think I'm missing something when really... the second condition seems irrelevant?